packages/shadycss/shadow-parts.md

```diff
@@ -0,0 +1,307 @@
+
+#### 1. Type selector is required
+
+Unlike the spec, `::part` rules <i>must</i> include a type selector (i.e. a
+custom element name). ShadyCSS relies on this hint to understand which scope
+each rule will flow to, without querying the DOM.
+
+<table>
+  <tr>
+    <th>Supported</th>
+    <th>Unsupported</th>
+  </tr>
+  <tr>
+    <td>
+      <code><b>x-a</b>::part(greeting)</code><br><br>
+      <code><b>x-a</b>#foo.bar::part(greeting)</code>
+    </td>
+    <td>
+      <code>::part(greeting)</code><br><br>
+      <code>#foo.bar::part(greeting)</code>
+    </td>
+  </tr>
+</table>
+
+#### 2. Specificity is crude
+
+The cascade/specificity behavior of `::part` rules is much more crude than the
+spec. `::part` rules are transformed to be `!important`, and otherwise there is
+no guarantee about precedence (actual behavior will depend largely on the order
+in which templates are processed).
+
+ShadyCSS rank (high to low):
+
+1. parent, grandparent, etc. scopes (no guarantee about order)
+2. local scope (if marked `!important`, local rules join group above)
+
+Spec rank (high to low):
+
+1. `!important` grandparent scope
+2. `!important` parent scope
+3. `!important` local scope
+4. grandparent scope
+5. parent scope
+6. local scope
+
+To workaround this limitation, you can increase the rank of any rule by
+adding additional selectors, such as `#id` and `.class` selectors.
+
+#### 3. Recursion is crude
+
+Certain "recursive" DOM structures will behave unexpectedly.
+
+In the example below, the spec-compliant behavior would display the `1 2 3`
+sequence with colors `red green red`, but ShadyCSS will produce `red green
+green`.
+
+```html
+<x-a>
+  #shadow-root
+    <style>
+      x-a.odd::part(num) {
+        color: red;
+      }
+      x-a.even::part(num) {
+        color: green;
+      }
+    </style>
+
+    <x-a class="odd">
+      #shadow-root
+        <style> <!-- as above --> </style>
+        <span part="num">1</num>
+
+        <x-a class="even">
+          #shadow-root
+            <style> <!-- as above --> </style>
+            <span part="num">2</num>
+
+            <x-a class="odd">
+              #shadow-root
+                <span part="num">3</num>
+            </x-a>
+        </x-a>
+    </x-a>
+```
+
+This happens because the `::part` rules are hoisted to the document scope and
+represented as shown below, thus there is no "resetting" after the second rule
+has matched an ancestor:
+
+```css
+x-a x-a.odd[shady-part~=x-a:x-a:num] {
+  color: red;
+}
+x-a x-a.even[shady-part~=x-a:x-a:num] {
+  color: green;
+}
+```
+
+Workarounds will vary based on specific scenario. For the above example, a good
+workaround would be to use two unique part names: `numOdd`, `numEven`.
+
+## How it works
+
+The basic approach is to convert `::part` rules to attribute selectors that
+encode the _providing_ scope, the _receiving_ scope, and the part name. Each
+shadow host then discovers its parent scope, and applies the correct attribute
```

Comment:
Does this update when the element moves into a different shadowRoot? If not, suggest adding this as a limitation.